x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypeConverter.java

```diff
@@ -369,6 +370,9 @@ public static DataType commonType(DataType left, DataType right) {
             if (left == TEXT || right == TEXT) {
                 return TEXT;
             }
+            if (left == SEMANTIC_TEXT || right == SEMANTIC_TEXT) {
+                return TEXT;
```

Comment:
Note that if the two types are the same, we always return that type, so this is saying that only in case of mixed type operations, we expect `SEMANTIC_TEXT` to cast to `TEXT`.  It seems a little odd to me that `KEYWORD == SEMANTIC_TEXT` would have a common type of `TEXT`, but maybe that's expected?  If so, I think a comment would help here.

Reply:
I changed this so that the common type is `KEYWORD` in the end.